ext/pcre/php_pcre.c

```diff
@@ -2156,23 +2157,28 @@ static zend_string *php_pcre_replace_array(HashTable *regex, zval *replace, zend
 
 /* {{{ php_replace_in_subject
  */
-static zend_always_inline zend_string *php_replace_in_subject(zval *regex, zval *replace, zval *subject, size_t limit, size_t *replace_count)
+static zend_always_inline zend_string *php_replace_in_subject(
+	zend_string *regex_str, HashTable *regex_ht,
+	zend_string *replace_str, HashTable *replace_ht,
+	zend_string *subject, size_t limit, size_t *replace_count)
 {
 	zend_string *result;
-	zend_string *subject_str = zval_get_string(subject);
+	zend_string *subject_str = zend_string_dup(subject, 0);
```

Comment:
This is unnecessary, the replacement is not performed in place.

Reply:
For the regex_str branch a copy isn't necessary either if you drop the corresponding zend_string_release_ex below.